app/components/slack-context-menu.js

```diff
@@ -0,0 +1,165 @@
+(function () {
+	'use strict';
+	// Load in our dependencies
+	var gui = window.require('nw.gui');
+	var clipboard = gui.Clipboard.get();
+
+	var GOOGLE_SEARCH_URL = "https://www.google.com/#q=";
+
+	function isSelected(target) {
+		return target.selectionStart != target.selectionEnd;
+	}
+
+	function getSelectedValue(target) {
+		return target.value.substring(target.selectionStart, target.selectionEnd);
+	}
+
+	function removeSelectedValue(target) {
+		var selectionStart = target.selectionStart;
+		var selectedValue = target.value.substring(target.selectionStart, target.selectionEnd);
+		target.value = target.value.substring(0, target.selectionStart) + target.value.substring(target.selectionEnd);
+		target.setSelectionRange(selectionStart,selectionStart); // place cursor at original selectionStart
+		return selectedValue;
+	}
+
+	function pasteValue(target, value) {
+		var selectionStart = target.selectionStart + value.length;
+		target.value = target.value.substring(0, target.selectionStart) + value + target.value.substring(target.selectionEnd);
+		target.setSelectionRange(selectionStart,selectionStart); // place cursor at original selectionStart
+	}
+
+	function buildTextareaMenu(menu, target) {
+		var selected = isSelected(target);
+
+		// Select all
+		menu.append(new gui.MenuItem({
+			label: 'Select All',
+			click: function () {
+				target.select();
+			}
+		}));
+		// TODO: Undo
+		// TODO: Redo
+		// Cut
+		menu.append(new gui.MenuItem({
+			label: 'Cut',
+			enabled: selected,
+			click: function () {
+				clipboard.set(removeSelectedValue(target));
+			}
+		}));
+		// Copy
+		menu.append(new gui.MenuItem({
+			label: 'Copy',
+			enabled: selected,
+			click: function () {
+				clipboard.set(getSelectedValue(target));
+			}
+		}));
+		// Paste
+		menu.append(new gui.MenuItem({
+			label: 'Paste',
+			enabled: clipboard.get() != undefined && clipboard.get().trim() != '',
+			click: function () {
+				pasteValue(target, clipboard.get());
+			}
+		}));
+		// Delete
+		menu.append(new gui.MenuItem({
+			label: 'Delete',
+			enabled: selected,
+			click: function () {
+				removeSelectedValue(target);
+			}
+		}));
+		// Search Google
+		if (selected) {
+			var selectedValue = getSelectedValue(target);
+			var labelValue = selectedValue;
+			if (labelValue.length > 10) {
+				labelValue = labelValue.substring(0,10) + "...";
+			}
+			menu.append(new gui.MenuItem({
+				label: 'Search Google for "' + labelValue + '"',
+				click: function() {
+					gui.Shell.openExternal( GOOGLE_SEARCH_URL + encodeURIComponent(selectedValue) );
```

Comment:
Your spacing for invocation is inconsistent, please remove the wrapping spaces on the sides of the parens:

```js
gui.Shell.openExternal(GOOGLE_SEARCH_URL + encodeURIComponent(selectedValue));
```